test/support/job_helper.rb

```diff
@@ -1,18 +1,49 @@
 module JobHelper
-  def self.included(base)
+  class << self
     def job_result_queue=(other)
       @job_result_queue = other
     end
 
     def job_result_queue
-      @job_result_queue
+      @job_result_queue ||= []
     end
   end
 
+
   class SimpleJob
     include Sidekiq::Worker
     def perform(*args)
-      self.job_result_queue << args
+      JobHelper.job_result_queue << args
+    end
+  end
+
+  class LockedJob < SimpleJob
+    sidekiq_options unique: :until_executed,
+                    unique_args: ->(args) { [] }
+  end
+
+  class Mgr
+    def options
+      { queues: ['default']}
     end
   end
+
+  require 'sidekiq/processor'
+  def work_off_jobs(redis)
```

Comment:
would we want tests for retry and dead sets also?

Reply:
We do but this branch doesn't support dead / retry sets (and the dead / retry set branch doesn't have test helpers). I want to merge this branch to _add_ those tests.